sshd-core/src/main/java/org/apache/sshd/certificate/OpenSshCertificateBuilder.java

```diff
@@ -267,4 +275,38 @@ public OpenSshCertificate sign(KeyPair caKeypair, String signatureAlgorithm) thr
 
         return cert;
     }
+
+    /**
+     * Validate if certificate options are correct
+     *
+     * @param options Options
+     */
+    private void validateOptions(List<OpenSshCertificate.CertificateOption> options) {
+        if (options != null && !options.isEmpty()) {
+            // check if any duplicates
+            Set<String> names = new HashSet<>();
+            Set<String> duplicates = options.stream().filter(option -> !names.add(option.getName()))
+                    .map(option -> option.getName())
+                    .collect(Collectors.toSet());
+            if (!duplicates.isEmpty()) {
+                throw new IllegalArgumentException("Duplicate option: " + String.join(",", duplicates));
+            }
+        }
+    }
+
+    /**
+     * Lexically order certificate options by "name"
+     *
+     * @param  options Options
+     * @return         Lexically ordered options
+     */
+    private List<OpenSshCertificate.CertificateOption> lexicallyOrderOptions(
+            List<OpenSshCertificate.CertificateOption> options) {
+        if (options != null && !options.isEmpty()) {
+            return options.stream()
+                    .sorted(Comparator.comparing(OpenSshCertificate.CertificateOption::getName))
+                    .collect(Collectors.toList());
+        }
+        return options;
```

Comment:
I would return here `Collections.emptyList()`. That would ensure that the builder stores a copy of the parameter the user passed in, and that subsequent modifications of the original list can have no effect on the builder.